crates/edr_napi/src/provider.rs

```diff
@@ -204,16 +210,48 @@ impl Provider {
                 }
             })
             .map_err(|error| napi::Error::new(Status::GenericFailure, error.to_string()))
-            .map(|data| {
+            .and_then(|data| {
                 let solidity_trace = solidity_trace.map(|trace| SolidityTraceData {
                     trace,
                     contract_decoder: Arc::clone(&self.contract_decoder),
                 });
-                Response {
+
+                // Normalise JSON-RPC format for rpcDebugTrace
+                let data = if method_name == "debug_traceTransaction"
+                    || method_name == "debug_traceCall"
+                {
+                    let parsed_data: Result<DebugTraceResult, _> = match &data {
+                        Either::A(json_string) => {
+                            serde_json::from_str::<JsonRpcResponse<DebugTraceResult>>(json_string)
+                                .map(|r| r.result)
+                        }
+                        Either::B(json_value) => serde_json::from_value::<
+                            JsonRpcResponse<DebugTraceResult>,
+                        >(json_value.clone())
+                        .map(|r| r.result),
+                    };
```

Comment:
It is incredibly inefficient to:

1. serialize the result to JSON (string or object)
2. deserialize the result from JSON (string or object)
3. normalize result
4. serialize the result

Could you please investigate making this change one level up, before we first serialize the result? That would be in `edr_provider`

Reply:
I have moved the normalisation into `edr_provider`